pkg/kubelet/cm/pod_container_manager_linux.go

```diff
@@ -91,8 +91,10 @@ func (m *podContainerManagerImpl) EnsureExists(pod *v1.Pod) error {
 		}
 		// Dispatch events to subscribed event handlers, if any.
 		// TODO(CD): Ensure it's ok to do this before calling applyLimits
-		if reply, err = m.eventDispatcher.PreStartPod(pod, string(podContainerName)); err != nil {
-			return fmt.Errorf("failed to execute PreStartPod hook for: %v: %v", pod.Name, err)
+		if m.eventDispatcher != nil {
+			if reply, err = m.eventDispatcher.PreStartPod(pod, string(podContainerName)); err != nil {
```

Comment:
@flyingcougar sounds like a good idea. If we go that route let's try to do it behind `newEventDispatcher`. That function is already supposed to be our singleton reference accessor for everyone who needs the event dispatcher.